aptos-move/block-executor/src/executor.rs

```diff
@@ -791,6 +827,39 @@ where
             }
         }
 
+        let mut patched_finalized_group_writes =
+            Self::map_id_to_values_in_group_writes(finalized_group_writes, &latest_view);
+
+        if let Some(group_reads_needing_delayed_field_exchange) =
```

Comment:
this only contains the read we had, which might be different from the snapshot at that time (as other resources could've been changed). we need to "capture the snapshot" for these, in the same way we did for finalized_groups. 
(not sure if we need both finalized_groups and finalized_groups_for_reads_needing_exchange, or one joint finalized_group is enough)